pkg/asset/machines/azure/machines.go

```diff
@@ -213,7 +213,7 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 	}
 
 	managedIdentity := fmt.Sprintf("%s-identity", clusterID)
-	if platform.IsARO() || platform.CloudName == azure.StackCloud {
```

Comment:
It would be helpful if each of these changes were in a separate commit, with a commit message describing why we think it is no longer needed. Without that it's impossible to review without repeating all of that work.